packages/xo-web/src/xo-app/host/tab-stats.js

```diff
@@ -4,56 +4,64 @@ import Icon from 'icon'
 import React from 'react'
 import Tooltip from 'tooltip'
 import { Container, Row, Col } from 'grid'
-import { DEFAULT_GRANULARITY, fetchStats, SelectGranularity } from 'stats'
+import { DEFAULT_GRANULARITY, fetchStats, SelectGranularity, INTERVAL_BY_GRANULARITY } from 'stats'
 import { Toggle } from 'form'
 import { CpuLineChart, MemoryLineChart, PifLineChart, LoadLineChart } from 'xo-line-chart'
 
 export default class HostStats extends Component {
   state = {
     granularity: DEFAULT_GRANULARITY,
     useCombinedValues: false,
+    statsIsPending: false,
   }
 
-  loop(host = this.props.host) {
+  fetchHostStats = () => {
+    if (this.state.statsIsPending) {
+      return
+    }
+
+    const host = this.props.host
+
     if (this.cancel) {
-      this.cancel()
+      return
     }
 
     if (host.power_state !== 'Running') {
       return
     }
 
-    let cancelled = false
-    this.cancel = () => {
-      cancelled = true
+    if (this.props.statsOverview?.interval === INTERVAL_BY_GRANULARITY[this.state.granularity.granularity]) {
+      this.setState({
+        stats: this.props.statsOverview,
+        selectStatsLoading: false,
+      })
+      return
     }
 
+    this.setState({
+      statsIsPending: true,
+    })
     fetchStats(host, 'host', this.state.granularity).then(stats => {
-      if (cancelled) {
-        return
-      }
-      this.cancel = null
-
-      clearTimeout(this.timeout)
-      this.setState(
-        {
-          stats,
-          selectStatsLoading: false,
-        },
-        () => {
-          this.timeout = setTimeout(this.loop, stats.interval * 1000)
-        }
-      )
+      this.setState({
+        stats,
+        selectStatsLoading: false,
+        statsIsPending: false,
+      })
     })
   }
+
+  loop() {
```

Comment:
This is not really a loop anymore. Should we change the name?

Reply:
`startInterval`?

Reply:
`initFetchHostStats`? Just to be exactly consistent with the name `fetchHostStats`.